### ISSUE TITLE
Warn if a test uses async code in a synchronous test

### ISSUE THREAD

**Issue by lo1tuma** [lo1tuma/eslint-plugin-mocha#5]
This should be considered as a warning:
```js
it('foo', function () {
    anyAsyncFunction(function (error, result) {
        expect(result).to.be.ok;
    });
});

it('foo', function () {
    anyAsyncFunction()
        .then(function (result) {
            expect(result).to.be.ok;
        });
});
```

This should not be considered as a warning:
```js
it('foo', function (done) {
    anyAsyncFunction(function (error, result) {
        expect(result).to.be.ok;
        done();
    });
});

it('foo', function (done) {
    anyAsyncFunction()
        .then(function (result) {
            expect(result).to.be.ok;
            done();
        });
});

it('foo', function () {
    return anyAsyncFunction()
        .then(function (result) {
            expect(result).to.be.ok;
        });
});
```
